benchmarks/vector_db_compare/topology_tax.py

```
@@ -0,0 +1,111 @@
+        _require(observed_generation == generation, "system-bench generation changed")
+        samples, elapsed = cell.get("total_nanos"), cell.get("elapsed_nanos")
+        _require(isinstance(samples, list) and len(samples) == 1000 and all(_uint(sample, positive=True) for sample in samples), "system-bench raw samples are invalid")
+        workers = cell["concurrency"]
+        _require(_uint(elapsed, positive=True) and elapsed >= max(samples) and elapsed >= (sum(samples) + workers - 1) // workers, "system-bench elapsed time is invalid")
+        _require((metrics.get("p50_nanos"), metrics.get("p95_nanos"), metrics.get("p99_nanos")) == (_percentile(samples, 50), _percentile(samples, 95), _percentile(samples, 99)), "system-bench percentiles changed")
+        _require(math.isclose(metrics.get("qps", 0), 1_000_000_000_000 / elapsed, rel_tol=1e-12), "system-bench QPS changed")
+
+
+def summarize(single_paths: list[Path], native_paths: list[Path]) -> dict[str, Any]:
+    _require(len(single_paths) == len(native_paths) == 3, "topology-tax baseline requires three repetitions per topology")
+    paths = dict(zip(TOPOLOGIES, (single_paths, native_paths)))
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)